protocol/src/codec.rs

```diff
@@ -13,8 +15,10 @@ pub struct MessageCodec;
 impl Encoder<Frame> for MessageCodec {
     type Error = anyhow::Error;
 
-    fn encode(&mut self, item: Frame, dst: &mut BytesMut) -> Result<(), Self::Error> {
```

Comment:
I'm dubious about removing these. I know it equates to the same thing, but it makes `type Error` meaningless and could obfuscate problems down the track if the error or result types change.